tools/oeedger8r/CMakeLists.txt

```diff
@@ -6,22 +6,89 @@
 # do not emit to the current working directory, they always emit to
 # the location of the input file.
 
-find_program(ESY esy)
-if (NOT ESY)
-  message(FATAL_ERROR "Please check your esy installation")
-endif ()
+if (USE_PLATFORM_OCAML)
+  # Use ocaml compilers if available.
+  # Use ocaml compilers if available
+  find_program(OCAMLLEX ocamllex)
```

Comment:
Hi,

I still don't think this is the right way to go. You cite issues installing `esy`, but that's just the package manager.

If you want to use the "platform's OCaml" then I suggest installing the OCaml build tool we're currently using, [dune](https://github.com/ocaml/dune) via your package manager.

This PR is conflating replacing a cross-platform OCaml package manager with replacing the de factor build system.

We should not maintain two OCaml build systems. That sounds sooo painful. The CMake hack to build the edger8r here is terrible to maintain, and generates a lot of noise due to warnings in the Intel code it can't silence (but which the dune build is setup to ignore automatically).

I think if you want to support using the "platform OCaml" you should change this PR to search for `dune` in the path, and then use it directly instead of via `esy`. The `esy` package manager mostly exists for its first-party Windows support, because we do not have a viable alternative for OCaml on Windows.